### ISSUE TITLE
Unsupervised Contrastive Loss Function

### ISSUE THREAD

**Issue by Dante-Basile** [neurodata/ProgLearn#520]
### Background

A loss function suitable for unsupervised learning would contribute to extending ProgLearn's training domain to unsupervised settings in which labeled data are not available. 

The unsupervised contrastive loss function is described in the paper [SimCLR - A Simple Framework for Contrastive Learning of Visual Representations](https://arxiv.org/pdf/2002.05709.pdf) and implemented at this [repository](https://github.com/google-research/simclr). It leverages data augmentations such as cropping and color shifts to enable unsupervised image classification. 

### Implementation

Addition of folder `unsupcon`. This will store the unsupervised contrastive loss function along with the associated demos and data. The goal is to provide a proof of concept to motivate integration of the unsupervised contrastive loss function with the ProgLearn network. For demonstration purposes, the [ResNet50](https://keras.io/api/applications/resnet/#resnet50-function) architecture will be trained on the [CIFAR-10](https://www.tensorflow.org/api_docs/python/tf/keras/datasets/cifar10/load_data) dataset in an experiment analogous to the one described in SimCLR.

**Comment by rflperry** [comment]
Would recommend FTE/BTE experiments like in the [supcon loss PR](https://github.com/neurodata/ProgLearn/pull/518). I think the gain of unsupcon over supcon is being able to leverage large amounts of unlabeled data to approach the supervised accuracy levels.

But the SimCLR paper examines transfer learning performance, learning an unsupervised representation and then given labeled data a 1-layer sigmoid classifier. ProgLearn does the same thing, but learns a kNN classifier.
